database/db.py

```diff
@@ -0,0 +1,210 @@
+from flaskext.mysql import MySQL
+from pymysql.err import InternalError, OperationalError, ProgrammingError
+
+#Available functions:
+#--------------------
+#dbConfig(app, user, password, host = "classmysql.engr.oregonstate.edu", dbName = None)
+#createTables()
+#index(query)
+#add(query)
+#insertCustomer(f_name, l_name, email, password, customer_id = None)
+#insertRetailer(name, url, retailer_id = None)
+#insertProduct(name, brand, manuf_model, product_id = None)
+#insertProductFromRetailer(product_id, retailer_id)
+#getCustomers()
+#getProducts()
+#getRetailers()
+#getRetailers_Products()
+#findCustomer(email, password)
+
+class Database():
+
+    #To initialize the DB instance, you should only really need to pass the app, user, and password.
+    def __init__(self, app, user, password, host = "classmysql.engr.oregonstate.edu", dbName = None):
+        
+        self.app = app
+        self.mysql = MySQL()
+        self.mysql.init_app(self.app)
+
+        if(dbName == None):
+            dbName = user
+
+        try:
+            #Username = cs361_*username* (eg. cs361_bahorat)
+            app.config['MYSQL_DATABASE_USER'] = str(user)
+            #Password = Last 4 of your student number (eg. 1011)
+            app.config['MYSQL_DATABASE_PASSWORD'] = str(password)
+            #Host = classmysql.engr.oregonstate.edu
+            app.config['MYSQL_DATABASE_HOST'] = str(host)
+            #DbName = Defualts to same as username
+            app.config['MYSQL_DATABASE_DB'] = str(dbName)
+        except Exception as e:
+            print("Could not configure app to connect to database")
+            print(e)
+
+        
+
+
+    #Sets up initial tables
+    def createTables(self):
```

Comment:
Did you plan to run this on every single start up? If so where in the code would this be called? I don't see it, but maybe you had an idea for a follow up PR?

Reply:
That's a good point I didn't even think about. I just updated it to make it only run on start up if there's a mismatch in the number of tables in the database

Reply:
Awesome. I didn't think about it until the review. sorry i missed it during our RDD!